File: src/router/baseNode.js
/**
 * Created by igor on 29.03.17.
 */

"use strict";

class BaseNode {
    constructor (parent, options = {}) {
        this._depth = 0;
        this._parent = parent;
        this.idx = 0;
        if (parent) {
            this.idx = parent.length;
            this._depth = parent.depth + 1;
        }

        [
            this.break = false,
            this.async = false,
            this.dump = false
        ] = [
            options.break,
            options.async,
            options.dump
        ]
    }

    get depth () {
        return this._depth;
    }

    getParent () {
        return this._parent;
    }
}

module.exports = BaseNode;

File: src/router/baseNode.go
/**
 * Created by I. Navrotskyj on 19.08.17.
 */

package router

type BaseNode struct {
	_depth int16
	idx    int
	parent *Node
}

func (n *BaseNode) getDepth() int16 {
	return n._depth
}

func (n *BaseNode) setParentNode(parent *Node) {
	if parent != nil {
		n.parent = parent
		n.idx = parent.getLength()
		n._depth = parent._depth + 1
	}
}

func (n *BaseNode) GetParent() *Node {
	if n.parent != nil {
		return n.parent
	}
	return nil
}
